packages/console/src/pages/SignInExperience/components/ManageLanguageModal/LanguageEditor.tsx

```diff
@@ -50,34 +46,58 @@ const LanguageEditor = ({ selectedLanguageTag, onFormStateChange }: LanguageEdit
     }
   );
 
-  const formMethods = useForm<Translation>();
+  const defaultFormValues = useMemo(
+    () =>
+      customPhrase && Object.keys(customPhrase.translation).length > 0
+        ? customPhrase.translation
+        : emptyUiTranslation,
+    [customPhrase]
+  );
+
+  const formMethods = useForm<Translation>({
+    defaultValues: defaultFormValues,
+  });
 
   const {
     handleSubmit,
     reset,
-    formState: { isSubmitting, isDirty },
+    formState: { isSubmitting, isDirty, dirtyFields },
   } = formMethods;
 
   useEffect(() => {
-    onFormStateChange(isDirty);
-  }, [isDirty, onFormStateChange]);
+    /**
+     * Note: This is a workaround for dirty state checking,
+     * for the `isDirty` state does not work correctly when comparing form data with empty / undefined values.
+     * Reference: https://github.com/react-hook-form/react-hook-form/issues/4740
+     */
+    setIsCurrentCustomPhraseDirty(isDirty && Object.keys(dirtyFields).length > 0);
```

Comment:
nit: wondering if other forms have a similar issue, could extract as a util method?

Reply:
Currently, we only occur this issue here; maybe extract it the next time we need to share this logic.